File: zig/day03/day03.zig
const std = @import("std");
const lr = @import("program_memory.zig");

pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const allocator = arena.allocator();

    const programMemory = try lr.parseProgramMemoryFile(allocator, "data/day03/input.txt");
    const instructions = try lr.parseAllInstructions(allocator, programMemory);
    const sum = lr.sumInstructions(instructions);
    std.debug.print("{{ sum = {d} }}\n", .{sum});
}

File: clj/src/day03/day03.clj
(require '[day03.program-memory :as pm])

(def test-file "data/day03/input.txt")

(def program-memory
  (pm/parse-program-memory-file {:file-name test-file}))

(def instructions
  (pm/all-valid-instructions {:program-memory program-memory}))

(println {:sum (pm/sum-all-multiplications {:instructions instructions})})
